### PR TITLE
feat(konnect): add DataPlaneClientCertificate reconciler

### LINKED ISSUE #608: Konnect entities: Data Plane Client Certificates
### Problem statement

Introduce `KongDataPlaneClientCertificate`.

API details: https://kong-platform-api.netlify.app/konnect/control-planes-config/v2/openapi.yaml/#tag/DP-Certificates

### Proposed solution

- Introduce `KongDataPlaneClientCertificate` CRD in https://github.com/Kong/kubernetes-configuration
  - All Kong configuration CRDs will live in that repository to prevent import cycles between KGO and KIC
  - This CRD will contain a ControlPlane reference. Only references that refer to Konnect will result in reconciliation against Konnect.

- Implement a reconciler which will use that CRD.

- This reconciler should be able to be disabled via CLI flag / env variable.
  - There should exist only 1 CLI flag / env variable for all Kong configuration entities like Consumers, Services, Routes etc.

### Acceptance criteria

- [ ] KGO can create, delete and update DataPlane's client certificates in Konnect based on the CRD spec.
- [ ] Status fields of this CRD are filled with information such as
  - [ ] Indication whether the entity in Konnect has been created via a `Programmed` status condition

### PR DESCRIPTION
**Pull request by czeslavo**
**What this PR does / why we need it**:

Adds a reconciler for the `KongDataPlaneClientCertificate` entity.

**Which issue this PR fixes**

Fixes https://github.com/Kong/gateway-operator/issues/608.

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect significant changes

### CODE REVIEW

File: test/envtest/konnect_entities_dataplaneclientcertificate_test.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		}
+		return lo.ContainsBy(c.Status.Conditions, func(condition metav1.Condition) bool {
+			return condition.Type == conditions.KonnectEntityProgrammedConditionType &&
+				condition.Status == metav1.ConditionTrue
+		})
+	}, "KongDataPlaneClientCertificate's Programmed condition should be true eventually")
+
+	t.Log("Waiting for KongDataPlaneClientCertificate to be created in the SDK")
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
+		assert.True(c, factory.SDK.CACertificatesSDK.AssertExpectations(t))
```
Comment (pmalek) [inline test/envtest/konnect_entities_dataplaneclientcertificate_test.go:87]:
```suggestion
		assert.True(c, sdk.DataPlaneCertificatesSDK.AssertExpectations(t))
```

File: test/envtest/konnect_entities_dataplaneclientcertificate_test.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	t.Log("Setting up SDK expectations on KongDataPlaneClientCertificate deletion")
+	sdk.DataPlaneCertificatesSDK.EXPECT().DeleteDataplaneCertificate(mock.Anything, cp.GetKonnectStatus().GetKonnectID(), dpCertID).
+		Return(&sdkkonnectops.DeleteDataplaneCertificateResponse{}, nil)
+
+	t.Log("Deleting KongDataPlaneClientCertificate")
+	require.NoError(t, cl.Delete(ctx, createdCert))
+
+	t.Log("Waiting for KongDataPlaneClientCertificate to be deleted in the SDK")
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		assert.True(c, factory.SDK.CACertificatesSDK.AssertExpectations(t))
```
Comment (pmalek) [inline test/envtest/konnect_entities_dataplaneclientcertificate_test.go:99]:
```suggestion
		assert.True(c, sdk.DataPlaneCertificatesSDK.AssertExpectations(t))
```

File: controller/konnect/watch_kongdataplanecertificate.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+		},
+	}
+}
+
+func enqueueKongDataPlaneClientCertificateForKonnectControlPlane(cl client.Client) handler.MapFunc {
+	return func(ctx context.Context, obj client.Object) []reconcile.Request {
+		auth, ok := obj.(*konnectv1alpha1.KonnectAPIAuthConfiguration)
+		if !ok {
+			return nil
+		}
```
Comment (pmalek) [inline controller/konnect/watch_kongdataplanecertificate.go]: This should be a `KonnectGatewayControlPlane` no?

**Review by pmalek — CHANGES_REQUESTED** [review]

File: controller/konnect/watch_kongdataplanecertificate.go (hunk 3 above)
Comment (czeslavo) [reply controller/konnect/watch_kongdataplanecertificate.go]: Good catch 😓 Fixed.

**Review by pmalek — APPROVED** [review]

**Pull request merged**